### DISCUSSION TITLE
CT2 Toggle

### DISCUSSION THREAD

**Discussion by stevie445** [nathanmarlor/foxess_modbus#381] [Ideas]
Hi,
I was looking at the possibility of using CT2 to monitor my EV charger to prevent the Fox batteries discharging to the car (during off-peak times only).  My understanding (and I may be wrong) is that CT1 & CT2(negative value) are added together to determine how much is left to charge the Fox batteries.

Assuming this works, it would be quite straight forward to wire this up and activate CT2 through the inverter panel but....

I would like to only use this function only for off-peak charging.  Would it be possible to add an inverter toggle switch to CT2?

I have a Zappie for my EV and it needs 1.4kW from solar before it starts to charge.  By toggling CT2 off, I can use the fox batteries as a 'float' when it drops below the threshold for a short time (I have automations for Zappie that gives the operating window).  This would help reduce the constant switching off & on due to a minor short dip in solar energy.
At night, I don't want the inverter to see the off-peak charging at all, so I would toggle CT2 on.

Hope this makes sense 
Steve

**Comment by FozzieUK** [comment]
Hi Steve, unfortunately CT2 doesn’t quite work the way you describe - CT2 is an external generator and is always negative (or zero, it will not go positive) and most importantly It’s a load side effect so when added to the load it reduces the size of the load and any surplus is export. 

If you fitted it to the Zappi, it would firstly need to be reversed so it sees charge as generation but then the inverter sees the Zappi charge as load as well so you will have effectively doubled the problem as there isn’t a -ve generation to reduce what the load CT is reading.

You’ll be better either putting your zappi on a Henley block by the meter in front of the inverters Ct clamp or if possible move the Ct clamp so it doesn’t see the zappi load (the zappi still sees export and functions normally)
Other options i’m assuming you have changed the Zappi export limit to 100W so it only starts when there is 100W clean of surplus above the 1.4kW, you can increase the stop delay for Eco+ so short dips have less effect and/or watch the Zappi charge and increase/decrease your minsoc on grid if you want to use (or not use) your battery to fill the gaps but that might see you occasionally use grid power if a big cloud reduces solar below your house usage.

**Comment by stevie445** [comment]
Thanks Fozzie,
My thinking was that CT1 would let the inverter know what the total consumption is, and the batteries would offset against this.  CT2 would be connected to the Zappie power only, and because it is negative, it would take this away from the CT1 total (effectively making the Zappie invisible to the inverter).  e.g. 3kw on CT1, -1kW on CT2 (Zappie) = Net 2kW

**Reply by FozzieUK** [reply]
Steve,

Unfortunately not - CT1 acts like a balance to the batteries, in a normal single inverter home it sees negative flow as export, and any positive flow as load - it balances it's output to the load so that CT1 is at zero. If there is spare solar generation it will send this to the grid and it knows CT1 will be reporting a negative value of it's generation less the load.

In your setup as the Zappi is on the inside of CT1 the charge is seen as part of the house load.

For example if you have 1.5kw of solar, and 300W load then CT1 will be measuring a negative 1,200W (export) - n.b. i'm ignoring losses.

If your Zappi turns on at 1.4kw the inverter will be seeing a load of 300W+1400W = 1,700W and so will have an import of 200W which it needs to balance and it will use 200W from the batteries to fill that load gap.


In a system that has a second solar inverter being measured by CT2 and using the same example but with the second inverter generating 1.4kw the hybrid inverter will see a negative 1.4kw on CT2, it knows to deduct this from the load, so 300W - 1,400W = -1,100W and it will have a balance point of -1,100W, the hybrid inverter also has 1.5kw of generation so expects there will be a negative 2.6kw (export) passing through CT1 and no battery will be needed.


In that same example but instead rather than a second inverter using CT2 to monitor the 1.4kw Zappi load -

CT2 will measure the Zappi charge of -1400kw (which it think it is generation), the load is 300W and because the Zappi is on the inside of CT1, the actual load will be 1,700W - the inverter knows this is incorrect as it was expecting -1,100W (300W-1400W) and so assumes the load must have instantly increased by 2,800W. It will then use it's 1.5kw of solar and take 1.3kw from the batteries to balance which has the effect of exporting your batteries to the grid at 1.3kw [not at all what you want].

When the Zappi is seen by the Hybrid as part of the house load the only mechanism you have to stop draining the batteries is to set a force charge window or increase grid minSoC so the inverter cannot discharge the batteries - or the best way involves some wiring and to move the Zappi feed so it is on the grid side of CT1 where any load it has cannot be measured by CT1.

CT2 can only ever be an external generation and *must* be seen to reduce the load at CT1 - Hope that makes some sense!

**Reply by canton7** [reply]
As an alternative, it looks like you can integrate the zappi into HA? Then you can set up an automation to disable battery discharge when the zappi is on (e.g. by setting the work mode / min soc).

Failing that, you can probably get a wifi-connected current clamp (such as a Shelly), connect that to the Zappi output and home assistant,
